src/App.tsx

```diff
@@ -14,11 +14,93 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [goods, setgoods] = useState([...goodsFromServer]);
+  const [visible, setVisible] = useState(false);
+  const visibleGoods = [...goods];
+
+  const start = () => {
+    setVisible(true);
+  };
+
+  const reverse = () => {
+    setgoods([...goods].reverse());
+  };
+
+  const sortAbc = () => {
+    setgoods([...goods].sort((a, b) => a.localeCompare(b)));
+  };
+
+  const sortLength = () => {
+    setgoods([...goods].sort((a, b) => a.length - b.length));
+  };
+
+  const reset = () => {
+    setgoods([...goodsFromServer]);
+  };
+
+  return (
+    <div className="App">
+      {!visible && (
+        <button
+          className="button"
+          type="button"
+          onClick={start}
+        >
+          Start
+        </button>
+      )}
+
+      {visible && (
+        <>
+          <ul className="list">
+            {visibleGoods.map(good => (
```

Comment:
```suggestion
            {goods.map(good => (
```
Попробуй вот так. Мне кажется, что от этого ничего не поменяется. Если нет, то сори)